apps/distgo/cmd/dist/dist.go

```diff
@@ -68,99 +75,106 @@ func Run(buildSpecWithDeps params.ProductBuildSpecWithDeps, stdout io.Writer) er
 	}
 
 	buildSpec := buildSpecWithDeps.Spec
-	for _, currDistCfg := range buildSpec.Dist {
-		if currDistCfg.Info.Type() == params.RPMDistType {
-			osArchs := buildSpec.Build.OSArchs
-			expected := osarch.OSArch{OS: "linux", Arch: "amd64"}
-			if len(osArchs) != 1 || osArchs[0] != expected {
-				return fmt.Errorf("RPM is only supported for %v", expected)
-			}
-			if err := checkRPMDependencies(); err != nil {
-				return err
+
+	distTypeOrdering := []params.DistInfoType{params.BinDistType, params.SLSDistType, params.RPMDistType, params.DockerDistType}
```

Comment:
Hard-coding this logic here seems a bit off. I think that it would be better to have a separate function that orders the `buildspec.Dist` contents according to these rules.